components/time/src/zone/offset.rs

```diff
@@ -199,34 +210,54 @@ impl VariantOffsetsCalculator {
     #[inline]
     #[expect(clippy::new_ret_no_self)]
     pub const fn new() -> VariantOffsetsCalculatorBorrowed<'static> {
-        VariantOffsetsCalculatorBorrowed {
-            offset_period: crate::provider::Baked::SINGLETON_TIMEZONE_VARIANTS_OFFSETS_V1,
-        }
+        VariantOffsetsCalculatorBorrowed::new()
     }
 
-    icu_provider::gen_buffer_data_constructors!(() -> error: DataError,
-        functions: [
-            new: skip,
-            try_new_with_buffer_provider,
-            try_new_unstable,
-            Self,
-        ]
-    );
+    #[cfg(feature = "serde")]
+    #[doc = icu_provider::gen_buffer_unstable_docs!(BUFFER, Self::new)]
+    pub fn try_new_with_buffer_provider(
+        provider: &(impl icu_provider::buf::BufferProvider + ?Sized),
+    ) -> Result<Self, DataError> {
+        use icu_provider::buf::AsDeserializingBufferProvider;
+        {
+            Ok(Self {
+                offset_period: if let Ok(payload) = DataProvider::<TimezonePeriodsV1>::load(
+                    &provider.as_deserializing(),
+                    Default::default(),
+                ) {
+                    OffsetData::New(payload.payload)
+                } else {
+                    OffsetData::Old(
+                        DataProvider::<TimezoneVariantsOffsetsV1>::load(
+                            &provider.as_deserializing(),
+                            Default::default(),
+                        )?
+                        .payload,
+                    )
```

Comment:
Nit: It's important to have a test covering this. Please check how we did it with the likely subtags data in the 1.x code. An easy way is to just add a postcard file with the data in the tests directory of this crate.